src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactVersionVulnerabilityScore.java

```diff
@@ -0,0 +1,127 @@
+import com.sap.oss.phosphor.fosstars.model.value.CVSS;
+import com.sap.oss.phosphor.fosstars.model.value.ScoreValue;
+import com.sap.oss.phosphor.fosstars.model.value.SemanticVersion;
+import com.sap.oss.phosphor.fosstars.model.value.VersionRange;
+import com.sap.oss.phosphor.fosstars.model.value.Vulnerabilities;
+import com.sap.oss.phosphor.fosstars.model.value.Vulnerability;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * The scores assesses if the artifact has known vulnerabilities and the severity of those.
+ * <ul>
+ *   <li>{@link OssFeatures#ARTIFACT_VERSION}</li>
+ *   <li>{@link OssFeatures#VULNERABILITIES}</li>
+ * </ul>
+ */
+public class ArtifactVersionVulnerabilityScore extends FeatureBasedScore {
+
+  private static final double NO_CPE_NO_VULNERABILITY_SCORE_VALUE = 8;
+
+  /**
+   * Initializes a new score.
+   */
+  public ArtifactVersionVulnerabilityScore() {
+    super("Known vulnerabilities for an artifact of an open-source project",
+        VULNERABILITIES, ARTIFACT_VERSION);
+  }
+
+  @Override
+  public ScoreValue calculate(Value<?>... values) {
+    Value<String> versionValue = findValue(values, ARTIFACT_VERSION,
+        "Hey! Give me info about used artifact version!");
+    Value<Vulnerabilities> vulnerabilities = findValue(values, VULNERABILITIES,
+        "Hey! Give me info about vulnerabilities!");
+
+    if (versionValue.isUnknown() || versionValue.isNotApplicable()) {
+      return scoreValue(Score.MIN, versionValue, vulnerabilities)
+          .makeUnknown().withMinConfidence()
+          .explain("Unknown version value.");
+    }
+
+    if (vulnerabilities.isUnknown()) {
+      return scoreValue(Score.MIN, versionValue, vulnerabilities)
+          .makeUnknown().withMinConfidence()
+          .explain("Unknown vulnerabilities value.");
+    }
+
+    if (vulnerabilities.get().isEmpty()) {
+      // if somehow possible it should be verified that CPE mapping was successful
+      return scoreValue(NO_CPE_NO_VULNERABILITY_SCORE_VALUE, versionValue, vulnerabilities)
+          .confidence(8)
+          .explain("No vulnerabilities for project found.");
+    }
+
+    Optional<SemanticVersion> semVer = SemanticVersion.parse(versionValue.get());
+    if (!semVer.isPresent()) {
+      return scoreValue(Score.MIN, versionValue, vulnerabilities)
+          .makeUnknown().withMinConfidence()
+          .explain("Version value is not parsable (not in semantic version format).");
+    }
+
+    return calculateScoreValue(versionValue, vulnerabilities, semVer.get());
+  }
+
+
+  private ScoreValue calculateScoreValue(Value<String> versionValue,
+      Value<Vulnerabilities> vulnerabilities, SemanticVersion usedVersion) {
+
+    int penalty = 0;
+    int issues = 0;
+    int highSeverityIssues = 0;
+    int mediumSeverityIssues = 0;
+    int lowSeverityIssues = 0;
+
+    Set<Vulnerability> entries = vulnerabilities.get().entries();
+    for (Vulnerability entry : entries) {
```

Comment:
I also would like to move the nested for-loop into a separate method.
But the inner loop requires access to all counters declared at begin of the method.
If the inner part is extracted we need some object which holds the counter.
If this is fine for you I would change it accordingly.

Regarding the `continue` I guess we have different opinions 😉
I prefer to avoid use of `continue` and for the `if` to check for the positive if possible (as it easier to read and understand).
So instead of 
```java
if(!inRange) {
  continue;
}
// do what to do
....
```

I prefer to have
```java
if(inRange) {
  // do what to do
  ....
}
```

Reply:
> But the inner loop requires access to all counters declared at begin of the method.

Reply:
I personally avoid use of `continue` because the reader can miss it.
And in this case here it only can be used with the `if(!inRange)` which I also try to avoid as the `!` can also missed during reading the method.

However, this is just my personal opinion and as the `continue` is also used in other classes/methods I will align with the coding style of the library 🙂